### PR TITLE
Update secp256k1 to v4.0.1

### PR DESCRIPTION
**Pull request by PhilippLgh**

### CODE REVIEW

File: src/rlpx/ecies.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ import {
 
 function ecdhX(publicKey: Buffer, privateKey: Buffer) {
   // return (publicKey * privateKey).x
-  return ecdhUnsafe(publicKey, privateKey, true).slice(1)
```
Comment (PhilippLgh) [inline src/rlpx/ecies.ts:20]: I assume that this breaks the tests. The ecdhUnsafe method seems to be removed from v4. Needs investigation

Comment (PhilippLgh) [reply src/rlpx/ecies.ts:20]: @holgerd77 any idea?

**Comment by holgerd77** [comment]
Thanks for tackling this. There are still 4 tests failing, also make sure to have a look at the linting error, always a favorite to be forgotten by everyone. 😋 (we should also add these Husky pre-push linting check we have added on the other libraries - e.g. [here](https://github.com/ethereumjs/merkle-patricia-tree/commit/c2332e509b679a479946dc736b281f66166242cd) on MPT - turned out to become really a time and nerve saver.

File: src/rlpx/ecies.ts (hunk 1 above)
Comment (holgerd77) [reply src/rlpx/ecies.ts:20]: Here was some issue on ecdhUnsafe from Vinay before, not sure if this is related: https://github.com/cryptocoinjs/secp256k1-node/issues/138

Comment (holgerd77) [reply src/rlpx/ecies.ts:20]: Have posted a [question](https://github.com/cryptocoinjs/secp256k1-node/pull/139#issuecomment-628718319) on this over on the `secp256k1-node` library.

Comment (holgerd77) [reply src/rlpx/ecies.ts:20]: See answer over on the repo thread

File: src/rlpx/ecies.ts [hunk 2]
```diff
@@ -17,7 +17,14 @@ import {
...
+  function hashfn(x: Uint8Array, y: Uint8Array) {
+    const pubKey = new Uint8Array(33)
+    pubKey[0] = (y[31] & 1) === 0 ? 0x02 : 0x03
+    pubKey.set(x, 1)
+    return pubKey
+  }
+  // @ts-ignore
+  return Buffer.from(ecdh(publicKey, privateKey, { hashfn }, Buffer.alloc(33)).slice(1))
 }
```
Comment (holgerd77) [inline src/rlpx/ecies.ts:28]: Just totally unable to review here. Where did you get this code from respectively - if you wrote yourself - can you further explain?

Comment (holgerd77) [reply src/rlpx/ecies.ts:28]: Update: ah, just seeing your private messages on this, ok, looks good in this context! 😄

**Pull request merged**